Realm/TestUtils/RLMTestCase.m

```diff
@@ -64,7 +64,8 @@ static BOOL encryptTests(void) {
     static BOOL encryptAll = NO;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        if (getenv("REALM_ENCRYPT_ALL")) {
+        const char *str = getenv("REALM_ENCRYPT_ALL");
+        if (str && *str) {
```

Comment:
The standard thing to check for a boolean environment variable is that the env variable is set to any non-empty value, and we were previously checking for it being set to any value and `REALM_ENCRYPT_ALL='' sh build.sh test-osx` would result in encryptAll being set. I initially thought this would be needed to get the `-encryption` tests using the non-CI scheme, but then realized I could toggle isEnabled in the scheme settings.

Reply:
Since the API can't edit environment variable settings, that required manually adding it every time a new workflow was created (such as for adding new Xcode versions). I added `REALM_ENCRYPT_ALL` to the environment variables in Realm.xcscheme but left it disabled by default, and ci_post_clone now toggles it to enabled when applicable.

Reply:
ohh that's perfect, so the only environment values which we need to add manually will be AWS Credentials for server tests